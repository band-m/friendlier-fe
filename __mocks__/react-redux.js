jest.mock('react-redux', () => {
  return {
    __esModule: true,
    useDispatch: jest.fn(() => () => {}),
    useSelector: jest.fn((selector) => selector({ contacts: { contactList: [] }, auth: { user: '' } }))
  };
});
