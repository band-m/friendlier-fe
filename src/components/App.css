@media screen and (min-width: 800px) {
  .Main{
    display: grid;
    grid-template-columns: 1fr 2fr;
  }
}
