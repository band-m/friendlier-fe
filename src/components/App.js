import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Header from './Header/Header';
import Home from './Home/HomePage';
import About from './About/AboutPage';
import Contacts from './Contacts/ContactDetail/ContactDetail';
import Settings from './Settings/SettingsPage';
import Signup from './Login/Signup';

export default function App() {
  return (
    <Router>
      <Header />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/signup" component={Signup} />
        <Route path="/contacts" component={Contacts} />
        <Route path="/about" component={About} />
        <Route path="/settings" component={Settings} />
      </Switch>
      {/* <Route path='/contacts/:id' component={ContactDetail} />
        <Route path='/contacts/add' component={AddContact} /> */}
    </Router>
  );
}
