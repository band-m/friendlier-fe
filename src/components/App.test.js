import React from 'react';
import 'react-redux';
import { shallow } from 'enzyme';
import App from './App';

jest.mock('../services/request.js');
jest.mock('../workers/subscribe-push.js');
jest.mock('../workers/unsubscribe-push.js');

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
