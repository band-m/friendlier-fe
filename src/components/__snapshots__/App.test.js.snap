// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<Component>
  <Header />
  <Component>
    <Component
      component={[Function]}
      exact={true}
      path="/"
    />
    <Component
      component={[Function]}
      path="/login"
    />
    <Component
      component={[Function]}
      path="/signup"
    />
    <Component
      component={[Function]}
      path="/contacts"
    />
    <Component
      component={[Function]}
      path="/add"
    />
    <Component
      component={[Function]}
      path="/edit/:id"
    />
    <Component
      component={[Function]}
      path="/about"
    />
    <Component
      component={[Function]}
      path="/settings"
    />
  </Component>
  <Component
    component={[Function]}
    path="/contacts/:id"
  />
</Component>
`;
